src/features/shared-link-settings-modal/SharedLinkSettingsModal.js

```diff
@@ -15,8 +15,31 @@ import ExpirationSection from './ExpirationSection';
 import AllowDownloadSection from './AllowDownloadSection';
 import messages from './messages';
 
+import { PEOPLE_WITH_LINK, PEOPLE_IN_COMPANY, PEOPLE_IN_ITEM } from '../shared-link-modal/constants';
+
 import './SharedLinkSettingsModal.scss';
 
+/**
+ * Return the translation key based on the access level and whether the user can download
+ * @param {string} accessLevel one of 'peopleWithTheLink', 'peopleInYourCompany', or 'peopleInThisItem'
+ * @param {boolean} canDownload prop value for whether the user can currently download
+ *
+ * @return {string} message ID for the proper transloation (or empty string when nothing matches)
+ */
+function getAccessNoticeMessageId(accessLevel, canDownload) {
+    switch (accessLevel) {
+        case PEOPLE_WITH_LINK:
+            return canDownload ? 'withLinkViewDownload' : 'withLinkView';
```

Comment:
You can use messages.xyz format directly, so that it's a reference and not a string constant, which is duplicated and harder to refactor. Also can try to minimize returns so that while debugging we aren't putting a break points in multiple lines vs just one or two returns to see what the function evaluates.

message = messages.inItem;
...

return message;
// OR return <FormattedMessage {...message}>